pkg/networkservice/common/interpose/server.go

```diff
@@ -135,10 +131,23 @@ func (l *interposeServer) Request(ctx context.Context, request *networkservice.N
 
 func (l *interposeServer) getConnectionID(conn *networkservice.Connection) string {
 	id := ""
-	for i := conn.GetPath().GetIndex(); i > 0; i-- {
-		if conn.GetPath().GetPathSegments()[i].Name == l.name {
-			id = conn.GetPath().GetPathSegments()[i].Id
+	segmentIdx := conn.GetPath().GetIndex()
+	isCrossNSEPrev := false
+
+	l.endpoints.Range(func(key string, crossNSEURL *url.URL) bool {
+		isCrossNSEPrev =
+			conn.Path.PathSegments[segmentIdx-1].Name == strings.TrimPrefix(key, interpose.InterposeNSEName)
```

Comment:
You should just check for Path segment Name to match endpoint name, you should not use interpose.InterposeNSENam it is just for registry.